aiida_registry/test_install.py

```diff
@@ -149,12 +155,44 @@ def test_install_one_docker(container_image, plugin):
     )
 
 
+def filter_entry_points(process_metadata, entrypoints):
+    """
+    Extract entry points that belongs to the plugin,
+    and delete any other entry points.
+    """
+
```

Comment:
If I understand correctly, here you are first building up the list of eps, then a list of keys to delete, then delete the keys

how about just _taking_ the entry points you want?

the code below is untested and likely not correct (I don't fully understand the structure of the current dictionaries), but it points in the direction
```
process_metadata = {}
for group_label, group_val in ENTRY_POINT_GROUPS:
  process_metadata[group_label]={}
  for ep_label, _ in process_metadata[group_label].items():
    process_metadata[group_label][ep_label] = group_val[ep_label]
```